File: MaglevMetroScorepad/app/src/main/java/brad/maglevmetroscorepad/GameHistory.java
package brad.maglevmetroscorepad;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.widget.TextView;

public class GameHistory extends AppCompatActivity {

    TextView display;

    String LIST = "list";
    String SHARED_PREFS = "sharedPrefs";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_game_history);

        display = findViewById(R.id.tv_gameHistory);

        load();
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        switch (item.getItemId()){
            case R.id.clear_history: clearGameHistory();
            default: break;
        }
        return super.onOptionsItemSelected(item);
    }

    public void clearGameHistory() {
        SharedPreferences sp = getSharedPreferences(SHARED_PREFS, MODE_PRIVATE);
        sp.edit().clear().commit();
        display.setText("");
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.game_history, menu);
        return true;
    }

    public void load(){
        SharedPreferences sharedPreferences = getSharedPreferences(SHARED_PREFS, MODE_PRIVATE);
        String[] res = sharedPreferences.getString(LIST, "").split(",", 0);
        for(int i=0; i<res.length; i++){
            display.append(res[i]+"\n");
        }
    }
}

File: CoalBarontheGreatCardGameScorepad/app/src/main/java/com/bradley/coalbaronthegreatcardgamescorepad/gameHistory.kt
package com.bradley.coalbaronthegreatcardgamescorepad

import android.content.SharedPreferences
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.TextView

import java.util.ArrayList

abstract class gameHistory : AppCompatActivity() {

    internal abstract var display: TextView
    internal var gameplay = ArrayList<String>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.game_history)
        display = findViewById(R.id.textView)
        //Load List
        load()
        //Print List
        displayList()
    }

    fun displayList() {
        val sb = StringBuilder()
        for (i in gameplay.indices) {
            sb.append(gameplay[i]).append("\n")
        }
        display.text = sb.toString()
    }

    private fun load() {
        val settings = getSharedPreferences("PREFS", 0)
        val items = settings.getString("Array", "")
        val arr = items!!.split(",".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        val arrayList = ArrayList<String>()
        for (i in arr.indices) {
            arrayList.add(arr[i])
        }
        gameplay = arrayList
    }
}